Jenkins slave

klinki opened the pull request:
Add jenkins-slave Dockerfile and configuration

jsight commented:
To get this working:

 - Add "xvfb" to the list of things installed
 - Add '--no-sandbox' to the list of chrome options
 - Update the Dockerfile to automatically start "Xvfb" along with the current options and export display based upon how it is run. In my case, I ran "Xvfb :1 -screen 5 1024x768x8" and "export DISPLAY=:1.5". 

I did this manually in a container and it seemed to work.

klinki commented:
I tried to do something very similar but Chrome kept crashing :( I will try it again...

This is what I did: 
```
RUN dnf install -y Xvfb libXfont Xorg \
    && /usr/bin/Xvfb :99 -ac -screen 0 1920x1080x24 &
RUN export DISPLAY=:99
```

edit: I see I cannot run processes in background this way. But I tried to use `docker exec -it $id-of-container bash` to connect to running container and to manually start  `Xvfb` and try protractor, but I keep getting these errors: 

> [12:59:45] I/testLogger - [chrome #01-1] PID: 448
> [chrome #01-1] Specs: /protractor_example/specs/nonAngularLoginSpec.js
> [chrome #01-1] 
> [chrome #01-1] [12:58:45] I/direct - Using ChromeDriver directly...
> [chrome #01-1] 
> [chrome #01-1] /protractor_example/node_modules/selenium-webdriver/lib/error.js:27
> [chrome #01-1]     super(opt_error);
> [chrome #01-1]     ^
> [chrome #01-1] WebDriverError: unknown error: Chrome failed to start: crashed
> [chrome #01-1]   (Driver info: chromedriver=2.30.477691 (6ee44a7247c639c0703f291d320bdf05c1531b57),platform=Linux 4.11.6-201.fc25.x86_64 x86_64)
> [chrome #01-1]     at WebDriverError (/protractor_example/node_modules/selenium-webdriver/lib/error.js:27:5)
> [chrome #01-1]     at Object.checkLegacyResponse (/protractor_example/node_modules/selenium-webdriver/lib/error.js:505:15)
> [chrome #01-1]     at parseHttpResponse (/protractor_example/node_modules/selenium-webdriver/lib/http.js:509:13)
> [chrome #01-1]     at doSend.then.response (/protractor_example/node_modules/selenium-webdriver/lib/http.js:440:13)
> [chrome #01-1]     at process._tickCallback (internal/process/next_tick.js:109:7)
> [chrome #01-1] From: Task: WebDriver.createSession()
> [chrome #01-1]     at Function.createSession (/protractor_example/node_modules/selenium-webdriver/lib/webdriver.js:777:24)
> [chrome #01-1]     at Function.createSession (/protractor_example/node_modules/selenium-webdriver/chrome.js:709:29)
> [chrome #01-1]     at Direct.getNewDriver (/protractor_example/node_modules/protractor/lib/driverProviders/direct.ts:90:25)
> [chrome #01-1]     at Runner.createBrowser (/protractor_example/node_modules/protractor/lib/runner.ts:225:39)
> [chrome #01-1]     at q.then.then (/protractor_example/node_modules/protractor/lib/runner.ts:391:27)
> [chrome #01-1]     at _fulfilled (/protractor_example/node_modules/q/q.js:834:54)
> [chrome #01-1]     at self.promiseDispatch.done (/protractor_example/node_modules/q/q.js:863:30)
> [chrome #01-1]     at Promise.promise.promiseDispatch (/protractor_example/node_modules/q/q.js:796:13)
> [chrome #01-1]     at /protractor_example/node_modules/q/q.js:556:49
> [chrome #01-1]     at runSingle (/protractor_example/node_modules/q/q.js:137:13)
> 
> [12:59:45] I/testLogger -

jsight commented:
@klinki - Try running chrome manually. In my case, this made me realize that I needed the "--no-sandbox" parameter for some reason. It would be better not to have it, but it didn't crash once that was there.

klinki commented:
@jsight Yep, it seems to be working with it. There is one last problem - how to start `xvfb` in docker.  I updated PR with e2e tests and added script which will start it. I hope it will work. 
(but there is problem with running it on desktop environment, I'm not sure if it would behave correctly)
